From d8a4b95be56f82221c674c04e57ca3039ff89261 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 20 Mar 2024 09:06:31 +0200
Subject: [PATCH] [SW-156330] gaudi3: increase ppboot size to 18kb

Current ppboot image size is very close to the 17kB
limit and addition of new code results in build
failure. Increasing the ppboot size limit from
to from 17kB to 18kB to allow new code addition.

Given that secondary image support is disabled,
PCIe link up time with this increase of ppboot
image size is calculated to be 77ms and well
within the 100ms limit for ppboot size 18kB.

Change-Id: I336a9442d705d66cad345731cb8f4f23a10c28bc
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index 9b5ee54..c3e02e0 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -1,7 +1,7 @@
 include $(HL_DEV_ARCH_COMMON)/Makefile
 include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
 
-PPBOOT_MAX_SIZE := 17408
+PPBOOT_MAX_SIZE := 18432
 
 ifeq ($(HL_TARGET),onix)
 	LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports_onix.elf
diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index c67f174..f8a3cfe 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -8,7 +8,7 @@
 
 #include <sizes.h>
 
-#define PPBOOT_SIZE			0x4400		/* 17KB */
+#define PPBOOT_SIZE			0x4800		/* 18KB */
 #define PREBOOT_SIZE			0x20000		/* 128KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x60000
